internal/core/services/room_manager/room_manager.go

```diff
@@ -98,10 +98,24 @@ func (m *RoomManager) DeleteRoom(ctx context.Context, gameRoom *game_room.GameRo
 		return fmt.Errorf("failed to delete instance on the runtime: %w", err)
 	}
 
-	err = m.SetRoomStatus(ctx, gameRoom, game_room.GameStatusTerminating)
+	terminatingGameRoom := *gameRoom
+	terminatingGameRoom.Status = game_room.GameStatusTerminating
+	err = m.setRoomStatus(ctx, gameRoom, &terminatingGameRoom)
 	if err != nil {
 		return fmt.Errorf("failed to update game room status to terminating: %w", err)
 	}
 
 	return nil
 }
+
+func (m *RoomManager) UpdateRoom(ctx context.Context, gameRoom *game_room.GameRoom) error {
+	currentGameRoom, err := m.roomStorage.GetRoom(ctx, gameRoom.SchedulerID, gameRoom.ID)
+	if err != nil {
+		return fmt.Errorf("unable to fetch game room from storage: %w", err)
```

Comment:
@gabrielcorado @luizmiranda7 I was inspired by the above functions' error handling when making the code of lines 114 and 118. Do you guys think we should be dealing with those errors differently?

Reply:
Basically, you're encapsulating the error within another, when checking types, if you're planning to handle this error by its type, this wrapping could be a problem. Otherwise, LGTM.

Reply:
Not sure if we will need to deal with those specific errors yet, probably going to find out in the next PR. Going to leave those as-is for now.